### DISCUSSION TITLE
[Feature]: Device scan for unique images should not lock up UI usage

### DISCUSSION THREAD

**Discussion by assid2** [immich-app/immich#1647] [Feature Request]
### Feature detail

Target Device: `Android`

Device scan for unique images currently locks up UI.

1. Use an old device with atleast a few thousand images in different folders , such as recent, whatsapp, camera, etc. 
2. When you go to choose the albums to backup, the UI is locked, as it scans for unique images at run time. 

Recommendation:
I suggest, you show up the folders list but do not scan at that time, once the user selects the Albums to backup, then run a seperate thread to scan for unique images in the background,  with a status on top saying "Scanning for Unique images", thereby not locking up the UI. The user can then continue setting up background/ foreground  etc  while not waiting for the scan in the background. After which the background backup can take place without intervention from the user.

### Platform

Mobile App

**Comment by alextran1502** [comment]
solve this will also solve #834

**Comment by ikaruswill** [comment]
My phone has only 7000 photos from my camera app but in total 40,000 photos due to messaging apps like WhatsApp, where their media folders are being picked up by immich.

I set up everything including OIDC but this has so far been a blocker, and I'm unable to get to the album selection even after leaving my phone screen on in hopes that the media scan will finish. 

Disabling the media detection with `.nomedia` in those messaging app folders is also not an option as I depend on those images being quickly available when I share media between messaging apps.

Would really appreciate if this can be looked into, I'd help out if I could in any way, but I'm not experienced in the languages in use here.
